liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -136,6 +141,31 @@ plugin = GHC.defaultPlugin {
                 failM
               Right newGblEnv' ->
                 pure newGblEnv'
+        | otherwise -> do
+            -- We're being called from the typecheckModuleIO call in typecheckedHook, so we avoid looping
+            pure gblEnv
+      where
+        thisModule = tcg_mod gblEnv
+
+--------------------------------------------------------------------------------
+-- | Inter-phase communication -------------------------------------------------
+--------------------------------------------------------------------------------
+--
+-- Since we have no good way of passing extra data between different
+-- phases of our plugin, we resort to horrible global mutable state.
+
+data Breadcrumb
+  = Parsed ParsedModule
+  | Typechecking
+
+breadcrumbs :: IORef (HM.HashMap StableModule Breadcrumb)
```

Comment:
Any reason to use `HashMap StableModule` rather than `Map ModuleName`. The later is simpler, while being reasonably efficient, I'd hope.

The stability of `StableModule` seems to matter only across different calls to GHC.

Reply:
Just because we already had `HashMap` in scope. I'll change it to use `Map ModuleName`.

Reply:
Is `ModuleName` unambiguous enough? You mentioned using the typechecker as a long-running server. In that case, can there be multiple units checked with the same server? If yes, you can encounter the same `ModuleName` in multiple units.

Reply:
(I realize `StableModule` is actually a `StableModuleName` so we're no worse off using `ModuleName` here, but still.)

Reply:
It was my slip. I think `Module` is the best choice.